ui/src/utils/fetcherHooks/pipelineViewFetch.ts

```diff
@@ -167,79 +195,88 @@ export const usePipelineViewFetch = (
     const metricsErr: any[] = [];
 
     if (spec?.vertices && vertexPods.size > 0) {
-      Promise.allSettled(
-        spec.vertices.map((vertex: any) => {
-          return fetch(`${BASE_API}/vertices/${vertex.name}/metrics`)
-            .then((response) => {
-              if (response.ok) {
-                return response.json();
-              } else {
-                return Promise.reject({ response, vertex: vertex.name });
-              }
-            })
-            .then((json) => {
-              const vertexMetrics = {
-                ratePerMin: "0.00",
-                ratePerFiveMin: "0.00",
-                ratePerFifteenMin: "0.00",
-                podMetrics: [],
-                error: false,
-              } as VertexMetrics;
-              let ratePerMin = 0.0,
-                ratePerFiveMin = 0.0,
-                ratePerFifteenMin = 0.0;
-              // keeping processing rates as summation of pod values
-              json.forEach((pod: any) => {
-                if ("processingRates" in pod) {
-                  if ("1m" in pod["processingRates"]) {
-                    ratePerMin += pod["processingRates"]["1m"];
-                  }
-                  if ("5m" in pod["processingRates"]) {
-                    ratePerFiveMin += pod["processingRates"]["5m"];
-                  }
-                  if ("15m" in pod["processingRates"]) {
-                    ratePerFifteenMin += pod["processingRates"]["15m"];
-                  }
-                } else {
-                  if (
-                    vertexPods.has(vertex.name) &&
-                    vertexPods.get(vertex.name) !== 0
-                  ) {
-                    vertexMetrics.error = true;
-                    metricsErr.push({
-                      error: `404: Failed to get metrics for ${vertex.name} vertex`,
-                      options: {
-                        toastId: `${vertex.name}-metrics`,
-                        autoClose: 5000,
-                      },
-                    });
+      Promise.allSettled([
+        fetch(`${BASE_API}/vertices/metrics`)
+          .then((response) => {
+            if (response.ok) {
+              return response.json();
+            } else {
+              return Promise.reject(response);
+            }
+          })
+          .then((json) => {
+            if (json?.data) {
```

Comment:
Maybe methodize some of this if possible. Or add some comments.

Reply:
added comments